wemake_python_styleguide/logic/tree/classes.py

```diff
@@ -43,39 +45,85 @@ def is_forbidden_super_class(class_name: Optional[str]) -> bool:
+    """Returns all non annotated class attributes of a class."""
     class_attributes = []
-    instance_attributes = []
+    for sub in ast.walk(node):
+        correct_context = nodes.get_context(sub) == node
+        has_value = getattr(sub, 'value', None)
+        if isinstance(sub, AssignNodes) and has_value and correct_context:
+            class_attributes.append(sub)
+
+    return class_attributes
 
+
+def get_all_class_attributes(
+    node: ast.ClassDef,
+) -> List[Union[ast.Assign, ast.AnnAssign]]:
+    """Returns all class attributes of a class."""
+    class_attributes = []
     for sub in ast.walk(node):
-        if isinstance(sub, ast.Attribute) and isinstance(sub.ctx, ast.Store):
-            instance_attributes.append(sub)
+        correct_context = nodes.get_context(sub) == node
+        has_value = getattr(sub, 'value', None)
+        if isinstance(sub, AssignNodes) and has_value and correct_context:
+            class_attributes.append(sub)
             continue
-
-        has_assign = (
-            nodes.get_context(sub) == node and
-            getattr(sub, 'value', None)
-        )
-        if isinstance(sub, AssignNodes) and has_assign:
+        if isinstance(sub, ast.AnnAssign) and correct_context:
             class_attributes.append(sub)
 
-    return class_attributes, instance_attributes
+    return class_attributes
+
+
+def get_instance_attributes(node: ast.ClassDef) -> List[ast.Attribute]:
+    """Returns all instance attributes of a class."""
+    instance_attributes = []
+    for sub in ast.walk(node):
+        if isinstance(sub, ast.Attribute) and isinstance(sub.ctx, ast.Store):
+            instance_attributes.append(sub)
+    return instance_attributes
 
 
-def getter_setter_postfixes(node: ast.ClassDef) -> Set[str]:
+def get_set_postfixes(node: ast.ClassDef) -> Tuple[Set[str], Set[str]]:
     """
-    Return postfixes of all getter or setter methods.
+    Return postfixes of class getter or setter methods.
 
     get_class_attribute becomes class_attribute
 
     set_instance_attribute becomes instance_attribute
 
     """
-    method_postfixes = set()
+    class_method_postfixes = set()
+    instance_method_postfixes = set()
     for sub in ast.walk(node):
-        correct_context = nodes.get_context(sub) == node
-        if isinstance(sub, FunctionNodes) and correct_context:
-            if any(sub.name.startswith(prefix) for prefix in ('get_', 'set_')):
-                method_postfixes.add(sub.name.partition('get_')[2])
-                method_postfixes.add(sub.name.partition('set_')[2])
-    return method_postfixes
+        if isinstance(sub, FunctionNodes) and functions.is_get_set(node, sub):
+            if functions.check_decorator(sub, 'classmethod'):
+                class_method_postfixes.add(sub.name.partition('get_')[2])
+                class_method_postfixes.add(sub.name.partition('set_')[2])
+                continue
+            instance_method_postfixes.add(sub.name.partition('get_')[2])
```

Comment:
I would better create two variables from `sub.name.partition('get_')[2]` and `sub.name.partition('set_')[2]`